packages/bpk-component-price-range/src/BpkPriceRange.tsx

```diff
@@ -0,0 +1,154 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useRef, useState } from 'react';
+
+import clamp from 'lodash/clamp';
+
+import BpkText, { TEXT_STYLES } from '../../bpk-component-text/src/BpkText';
+import { cssModules } from '../../bpk-react-utils';
+
+import BpkPriceMarker from './BpkPriceMarker';
+import { MARKER_TYPES } from './common-types';
+
+import type { MarkerType } from './common-types';
+
+import STYLES from './BpkPriceRange.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+type Marker = {
+  price: string;
+  percentage: number;
+};
+
+export type Props = {
+  min?: number;
+  max?: number;
+  showPriceIndicator?: boolean;
+  marker: Marker;
+  segments: {
+    low: Marker;
+    high: Marker;
+  };
+};
+
+const BpkPriceRange = ({
+  marker,
+  max = 100,
+  min = 0,
+  segments,
+  showPriceIndicator = true,
+}: Props) => {
+  const linesRef = useRef<HTMLDivElement>(null);
+  const indicatorRef = useRef<HTMLDivElement>(null);
+  const [prefilledWidth, setPrefilledWidth] = useState(0);
+  const calcPercentage = (current: number) =>
+    (clamp(current, min, max) - min) / (max - min);
+  const indicatorPercent = calcPercentage(marker.percentage);
+
+  let type: MarkerType;
+  if (marker.percentage < segments.low.percentage) {
+    type = MARKER_TYPES.LOW;
+  } else if (marker.percentage > segments.high.percentage) {
+    type = MARKER_TYPES.HIGH;
+  } else {
+    type = MARKER_TYPES.TYPICAL;
+  }
+
+  useEffect(() => {
+    // to calculate the spacing ahead of the price indicator
+    if (indicatorRef.current && linesRef.current) {
+      const estimatedWidth =
+        indicatorPercent * linesRef.current.scrollWidth -
+        indicatorRef.current.scrollWidth / 2;
+      const maxPrefilledWidth =
+        linesRef.current.scrollWidth - indicatorRef.current.scrollWidth;
+      const actualPrefilledWidth = clamp(estimatedWidth, 0, maxPrefilledWidth);
+
+      setPrefilledWidth(actualPrefilledWidth);
+    }
+  }, [indicatorPercent]);
+
+  const linesClassNames = {
+    lines: getClassName(
+      'bpk-price-range__lines',
+      showPriceIndicator && 'bpk-price-range__lines--large',
+    ),
+    low: getClassName(
+      'bpk-price-range__line--low',
+      showPriceIndicator && 'bpk-price-range__line--lowLarge',
+    ),
+    high: getClassName(
+      'bpk-price-range__line--high',
+      showPriceIndicator && 'bpk-price-range__line--highLarge',
+    ),
+    typical: getClassName('bpk-price-range__line--typical'),
+    dot: getClassName(
+      `bpk-price-range__line--${type}`,
+      'bpk-price-range__line--dot',
+    ),
+  };
```

Comment:
To follow more the format of other Backpack components, I would change this to be in that format of

```suggestion
  const lines = getClassName(
      'bpk-price-range__lines',
      showPriceIndicator && 'bpk-price-range__lines--large',
    );
    const low = getClassName(
      'bpk-price-range__line--low',
      showPriceIndicator && 'bpk-price-range__line--lowLarge',
    );
    const high = getClassName(
      'bpk-price-range__line--high',
      showPriceIndicator && 'bpk-price-range__line--highLarge',
    );
    const medium = getClassName('bpk-price-range__line--typical');
    const dot = getClassName(
      `bpk-price-range__line--${type}`,
      'bpk-price-range__line--dot',
    );
```


 Equally you could also inline these to the `div` they are used below

Reply:
upadted, thanks